src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -190,9 +191,12 @@ export class TerminalService implements ITerminalService {
 
 		if (processCount === 0 && !name) {
 			name = this.nextTerminalName;
+			shellPath = this.nextTerminalShellPath;
```

Comment:
Create the config here so that the config isn't tracked in 2 vars:

```
private _nextTerminalConfiguration: ITerminalProcessConfiguration;

...

if (processCount === 0 && !name) {
  this._nextTerminalConfiguration = {
    name: ...
    shellPath: ...
  };
} else {
  ...
}
```